### PR TITLE
[FEATURE] make SDK support 4.8 (auto time series train and predict)

### PR DESCRIPTION
**Pull request by NoctisHong**
Signed-off-by: Noctis Hong <noctis@mobagel.com>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -113,6 +113,7 @@ venv/
 ENV/
 env.bak/
 venv.bak/
+tmp/
```
Comment (CliffLin) [inline .gitignore:116]: what is it?

File: src/decanter/core/core_api/body_obj.py [hunk 2]
```diff
@@ -144,29 +143,39 @@ def create(cls, **kwargs):
 class TSGroupBy(CoreBodyObj):
     """Time series group by."""
     @classmethod
-    @corex_obj(
-        required={'time_unit', 'regression_method', 'classification_method'})
+    @corex_obj(required=None)
     def create(cls, **kwargs):
         """Return TSGroupBy object with passed kwargs as attributes"""
+        at_least = {
```
Comment (CliffLin) [inline src/decanter/core/core_api/body_obj.py:149]: check requirements with decorator, corex_obj.

File: src/decanter/core/core_api/body_obj.py [hunk 3]
```diff
@@ -182,14 +191,13 @@ def create(cls, **kwargs):
 
 
 class PredictBodyTSModel(CoreBodyObj):
-    """Body for predict time series model api."""
+    """Body for predict time series forcast model api."""
```
Comment (CliffLin) [inline src/decanter/core/core_api/body_obj.py:194]: forecast

File: src/decanter/core/core_api/body_obj.py [hunk 4]
```diff
@@ -144,29 +143,39 @@ def create(cls, **kwargs):
...
+    """Attribute for TrainAutoTSBody."""
+    @classmethod
+    @corex_obj(required={'genetic_algorithm'})
+    def create(cls, **kwargs):
+        """Return BuildSpec object with passed kwargs as attributes"""
+        return cls(**kwargs)
 
 class TrainAutoTSBody(CoreBodyObj):
-    """Body for train times series api."""
+    """Body for auto ts train api."""
```
Comment (CliffLin) [inline src/decanter/core/core_api/body_obj.py:176]: time series

File: src/decanter/core/core_api/train_input.py [hunk 5]
```diff
@@ -100,90 +100,93 @@ def get_train_params(self):
...
-            validation_percentage=validation_percentage,
-            max_model=max_model)
 
-        genetic_algorithm = CoreBody.GeneticAlgorithmParams.create(
+        if id is None:
+            columns = None
+        elif nullable != None:
+            if len(id) == len(data_type) and len(data_type) == len(nullable) :
+                columns = map(
+                        lambda tuple3:CoreBody.Column.create(id=tuple3[0], data_type=tuple3[1], nullable=tuple3[2]),
```
Comment (CliffLin) [inline src/decanter/core/core_api/train_input.py:126]: tuple3 is a wired variable name.
Is it possible to unpack it with `lambda id, data_type, nullable: ...., id`?

File: src/decanter/core/core_api/train_input.py [hunk 6]
```diff
@@ -100,90 +100,93 @@ def get_train_params(self):
...
+                        [(id[i], data_type[i], nullable[i]) for i in range(len(id))]
+                    )
+            else:
+                raise ValueError('id, data_type and nullable list length require identically.')
+        elif len(id) == len(data_type):
+            columns = map(
+                    lambda tuple2:CoreBody.Column.create(id=tuple2[0], data_type=tuple2[1]),
```
Comment (CliffLin) [inline src/decanter/core/core_api/train_input.py:133]: ditto

**Review by CliffLin — COMMENTED** [review]
please run make lint before going to commit code

File: README.md [hunk 7]
```diff
@@ -41,6 +41,14 @@ $ python -m example.file
 Progress UploadTask_train:  55%|█████████████████████████████████████████
 ```
 
+## Auto Time Series Example: Upload Data, Train TSF Experiment and Predict
```
Comment (CliffLin) [inline README.md:44]: please don't use abbreviation, TSF

File: src/decanter/core/core_api/train_input.py (hunk 5 above)
Comment (schwannden) [reply src/decanter/core/core_api/train_input.py:126]: What is this nullable? I can't seem to find it in CoreX API

File: src/decanter/core/core_api/body_obj.py (hunk 2 above)
Comment (NoctisHong) [reply src/decanter/core/core_api/body_obj.py:149]: Thanks, I'll fix that~

File: src/decanter/core/core_api/body_obj.py (hunk 3 above)
Comment (NoctisHong) [reply src/decanter/core/core_api/body_obj.py:194]: Good eyes

File: src/decanter/core/core_api/body_obj.py (hunk 4 above)
Comment (NoctisHong) [reply src/decanter/core/core_api/body_obj.py:176]: Good eyes

File: src/decanter/core/core_api/train_input.py (hunk 5 above)
Comment (NoctisHong) [reply src/decanter/core/core_api/train_input.py:126]: @schwannden
The nullable field is in the requestBody of `/v2/tasks/setup` -> `SetupBody` -> `data_columns*: [Column]` -> `(id*, data_type*, nullable)`

File: src/decanter/core/core_api/train_input.py (hunk 6 above)
Comment (NoctisHong) [reply src/decanter/core/core_api/train_input.py:133]: Got it

File: README.md (hunk 7 above)
Comment (NoctisHong) [reply README.md:44]: OK, I see.

File: .gitignore (hunk 1 above)
Comment (NoctisHong) [reply .gitignore:116]: allow developers or project use/create some file in tmp folder and prevent pushing them to repo.

**Review by schwannden — APPROVED** [review]

**Pull request merged**